### ISSUE TITLE
Signed distance doesn't work on BVH Models.

### ISSUE THREAD

**Issue by BryceStevenWilley** [flexible-collision-library/fcl#221]
According to the comment at https://github.com/flexible-collision-library/fcl/blob/master/include/fcl/narrowphase/distance-inl.h#L140, there should be a work around in place to calculate negative distance for BVH meshes (even if the closest points aren't correct), but in practice, even `enabling request.enable_signed_distance = true` doesn't calculate signed distance, and the returned min_distance is always 0. 

I've put together a test demonstrating this [here](https://github.com/BryceStevenWilley/fcl/blob/master/test/test_fcl_signed_distance.cpp).

**Comment by marc1706** [comment]
It seems to me like the workaround is actually never hit due an incorrect assumption at https://github.com/flexible-collision-library/fcl/blob/master/include/fcl/narrowphase/distance-inl.h#L148.

As the comment on line 140 states the min distance will be 0 when using BVH or octree. However, the check on line 148 will evaluate as false as 0 is clearly not less than `static_cast<S>(0)` and the workaround will therefore not be executed.

**Comment by BrettHemes** [comment]
@marc1706 I found that the test on `res` is also blocking.  I do not know the exact functionality of the lookup function but it seems to be returning 0 as well when min_distance is 0.

I added an additional <= to the if check in addition to that of #228 and was able to get negative distances returned.  The resulting if check is as follows:
```
if(res <= static_cast<S>(0)
   && result.min_distance <= static_cast<S>(0)
   && request.enable_signed_distance)
{
```

not sure if this is correct but it was the only way I could get it to work as expected.

**Comment by marc1706** [comment]
You're correct that res also seems to be blocking. The distance function seems to return `result.min_distance` by default. As a result of that checking `res` and `result.min_distance` is not needed. I've updated my pull request accordingly.

**Comment by wxmerkt** [comment]
There is a case where adding this new assumption can lead to a segmentation fault: ``res`` can return ``-1`` if the GJK solver cannot find an intersection - the check ``<= 0`` would go into the fake-nearest-point computation and then segfault because the ``assert`` statements are not obeyed in Release mode and an access to an invalid memory point (``pos``) made if the index is still as initialised.
